src/lib/optimizer/join_ordering/join_graph_converter.cpp

```diff
@@ -0,0 +1,192 @@
+#include "join_graph_converter.hpp"
+
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+JoinGraph JoinGraphConverter::operator()(const std::shared_ptr<AbstractLQPNode>& lqp) {
+  /**
+   * Traverse the LQP until the first non-vertex type (e.g. a UnionNode) is found or a node doesn't have precisely
```

Comment:
Could you expand on what makes an LQP node a "(non-)vertex" type? Maybe in `_is_vertex_type`?

Reply:
Is this maybe a `_type_is_relevant_for_join_graph`?

Reply:
called it _lqp_node_type_is_vertex and added a doc comment